src/aws-healthomics-mcp-server/awslabs/aws_healthomics_mcp_server/tools/annotation_store_tools.py

```diff
@@ -0,0 +1,380 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Annotation store tools for the AWS HealthOmics MCP server."""
+
+import botocore.exceptions
+from awslabs.aws_healthomics_mcp_server.consts import (
+    DEFAULT_MAX_RESULTS,
+)
+from awslabs.aws_healthomics_mcp_server.utils.aws_utils import (
+    get_omics_client,
+)
+from loguru import logger
+from mcp.server.fastmcp import Context
+from pydantic import Field
+from typing import Any, Dict, List, Optional
+
+
+async def list_aho_annotation_stores(
+    ctx: Context,
+    max_results: int = Field(
+        DEFAULT_MAX_RESULTS,
+        description='Maximum number of annotation stores to return',
+        ge=1,
+        le=100,
+    ),
+    next_token: Optional[str] = Field(
+        None,
+        description='Token for pagination from a previous response',
+    ),
+) -> Dict[str, Any]:
+    """List available HealthOmics annotation stores.
+
+    Args:
+        ctx: MCP context for error reporting
+        max_results: Maximum number of annotation stores to return
+        next_token: Token for pagination from a previous response
+
+    Returns:
+        Dictionary containing annotation stores list and pagination info
+
+    Raises:
+        Exception: If there's an error listing annotation stores
+    """
+    try:
+        client = get_omics_client()
+
+        params = {'maxResults': max_results}
+
+        if next_token:
+            params['nextToken'] = next_token
+
+        response = client.list_annotation_stores(**params)
+
+        return {
+            'annotationStores': response.get('annotationStores', []),
+            'nextToken': response.get('nextToken'),
+            'totalCount': len(response.get('annotationStores', [])),
+        }
+
+    except botocore.exceptions.ClientError as e:
+        error_code = e.response['Error']['Code']
+        error_message = e.response['Error']['Message']
+
+        logger.error(f'Failed to list annotation stores: {error_code} - {error_message}')
+
+        raise Exception(f'Failed to list annotation stores: {error_code} - {error_message}')
+    except Exception as e:
+        logger.error(f'Unexpected error listing annotation stores: {str(e)}')
+        raise Exception(f'Failed to list annotation stores: {str(e)}')
+
+
+async def get_aho_annotation_store(
+    ctx: Context,
+    annotation_store_id: str = Field(
+        ...,
+        description='ID of the annotation store to retrieve',
+    ),
+) -> Dict[str, Any]:
+    """Get detailed information about a specific annotation store.
+
+    Args:
+        ctx: MCP context for error reporting
+        annotation_store_id: ID of the annotation store
+
+    Returns:
+        Dictionary containing annotation store details
+
+    Raises:
+        Exception: If there's an error retrieving annotation store information
+    """
+    try:
+        client = get_omics_client()
```

Comment:
`get_omics_client()` currently creates a new connection every time. When called in every function in the aho `tools/`, a new connection is being created every time which creates overhead. Have we considered reusing the connections by perhaps client caching or connection pooling?